modules/core/src/test/java/org/apache/ignite/internal/processors/cache/msgtimelogging/CacheMessagesTimeLoggingTest.java

```diff
@@ -0,0 +1,240 @@
+import org.apache.ignite.internal.processors.cache.distributed.dht.GridDhtTxQueryEnlistResponse;
+import org.apache.ignite.internal.processors.cache.distributed.dht.GridDhtTxQueryFirstEnlistRequest;
+import org.apache.ignite.internal.processors.cache.distributed.dht.atomic.GridNearAtomicFullUpdateRequest;
+import org.apache.ignite.internal.processors.cache.distributed.dht.atomic.GridNearAtomicSingleUpdateRequest;
+import org.apache.ignite.internal.processors.cache.distributed.dht.atomic.GridNearAtomicUpdateResponse;
+import org.apache.ignite.internal.processors.cache.distributed.near.GridNearLockRequest;
+import org.apache.ignite.internal.processors.cache.distributed.near.GridNearLockResponse;
+import org.apache.ignite.internal.processors.cache.distributed.near.GridNearTxEnlistRequest;
+import org.apache.ignite.internal.processors.cache.distributed.near.GridNearTxEnlistResponse;
+import org.apache.ignite.internal.processors.cache.distributed.near.GridNearTxPrepareRequest;
+import org.apache.ignite.internal.processors.cache.distributed.near.GridNearTxPrepareResponse;
+import org.apache.ignite.internal.processors.metric.HistogramMetric;
+import org.apache.ignite.spi.communication.tcp.TcpCommunicationSpiMBean;
+import org.apache.ignite.transactions.Transaction;
+import org.junit.Test;
+
+import static org.apache.ignite.IgniteSystemProperties.IGNITE_COMM_SPI_TIME_HIST_BOUNDS;
+import static org.apache.ignite.IgniteSystemProperties.IGNITE_ENABLE_MESSAGES_TIME_LOGGING;
+import static org.apache.ignite.cache.CacheAtomicityMode.ATOMIC;
+import static org.apache.ignite.cache.CacheAtomicityMode.TRANSACTIONAL;
+import static org.apache.ignite.cache.CacheAtomicityMode.TRANSACTIONAL_SNAPSHOT;
+import static org.apache.ignite.cache.CacheWriteSynchronizationMode.FULL_SYNC;
+import static org.apache.ignite.spi.communication.tcp.TcpCommunicationMetricsListener.DEFAULT_HIST_BOUNDS;
+import static org.apache.ignite.transactions.TransactionConcurrency.PESSIMISTIC;
+import static org.apache.ignite.transactions.TransactionIsolation.READ_COMMITTED;
+
+/**
+ * Tests for CommunicationSpi time metrics.
+ */
+public class CacheMessagesTimeLoggingTest extends GridCacheMessagesTimeLoggingAbstractTest {
+    /**
+     *
+     */
+    @Test
+    public void testGridDhtTxPrepareRequestTimeLogging() throws MalformedObjectNameException {
+        IgniteCache<Integer, Integer> cache = grid(0).cache(DEFAULT_CACHE_NAME);
+
+        populateCache(cache);
+
+        checkOutcomingEventsNum(GridDhtTxPrepareRequest.class, GridDhtTxPrepareResponse.class);
+    }
+
+    /**
+     * Near node sends requests to primary node but gets responses from backup node
+     * for atomic caches with full sync mode.
+     * Time logging must be disabled for this case.
+     */
+    @Test
+    public void testAtomicFullSyncCache() throws MalformedObjectNameException {
+        IgniteCache<Integer, Integer> cache0 = grid(0).createCache(new CacheConfiguration<Integer, Integer>()
```

Comment:
Please move cache configuration to local variable.

Reply:
This construction is not uncommon and doesn't violate our code conventions. I prefer to keep it